lib/src/__tests__/typescript/GenericValues.tstest.tsx

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import * as React from 'react';
+import moment, { Moment } from 'moment';
+import LuxonAdapter from '@material-ui/pickers/adapter/luxon';
+import { DateTime } from 'luxon';
+import { TextField } from '@material-ui/core';
+import { DatePicker } from '@material-ui/pickers';
+import { DateRangePicker } from '../../DateRangePicker/DateRangePicker';
+
+// Allows to set date type right with generic JSX syntax
+<DatePicker<Date>
+  value={new Date()}
+  onChange={date => date?.getDate()}
+  renderInput={props => <TextField {...props} />}
+/>;
+
+// Also works for DateRangePicker
+<DateRangePicker<DateTime>
+  value={[new DateTime(), new DateTime()]}
+  onChange={date => date[0]?.set({ second: 15 })}
+  renderInput={props => <TextField {...props} />}
+/>;
+
+// Throws error if passed value is invalid
+<DatePicker<Date>
+  // @ts-expect-error Value is invalid
+  value={new DateTime()}
+  onChange={date => date?.getDate()}
+  renderInput={props => <TextField {...props} />}
+/>;
+
+// Inference from the state
+const InferTest = () => {
+  const [date, setDate] = React.useState<DateTime | null>(new DateTime());
+
+  return (
+    <DatePicker
+      value={date}
+      onChange={date => setDate(date)}
+      renderInput={props => <TextField {...props} />}
+    />
+  );
+};
+
+// Infer value type from the dateAdapter
+<DatePicker
+  value={new DateTime()}
+  onChange={date => console.log(date)}
+  renderInput={props => <TextField {...props} />}
+  dateAdapter={new LuxonAdapter()}
```

Comment:
From what I understand, the prop will almost never be used. I believe people will use the `LocalizationProvider` in +90% of the use cases. I don't think that we need to test it.

I also think that we should move toward removing this prop, starting by hiding it and seeing the feedback we get. I will open an RFC for it.